### PR TITLE
Some MeiliSearch v0.21.0 related changes

### PR DESCRIPTION
**Pull request by juzi5201314**
https://github.com/meilisearch/integration-guides/issues/117
change: 
- Ranaming of buildDate into commitDate
- New update status value
- attributesForFaceting is renamed into filterableAttributes
- Rename fieldsDistribution into fieldDistribution
- filters is renamed into filter
- facetFilters is removed

### CODE REVIEW

File: src/search.rs [hunk 1]
```diff
@@ -144,14 +144,10 @@ pub struct Query<'a> {
...
-    /// Filters applied to documents.
+    /// Filter applied to documents.
     /// Read the [dedicated guide](https://docs.meilisearch.com/reference/features/filtering.html) to learn the syntax.
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub filters: Option<&'a str>,
-    /// Facet names and values to filter on.
-    /// Read [this page](https://docs.meilisearch.com/reference/features/search_parameters.html#facet-filters) for a complete explanation.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub facet_filters: Option<&'a [&'a [&'a str]]>,
+    pub filter: Option<&'a str>,
```
Comment (bidoubiwa) [inline src/search.rs:150]: The type of fillter is the same as `facet_filters` but can also accept a `str` as value. So it's a mix of both `filter` and `facet_filters`. 
The following syntaxes are possible: 

nested array:
```json
["something > 1", "genre=comedy", ["genre=horror", "title=batman"]]
```
only string: 
```json
"something > 1 AND genre=comedy AND (genre=horror OR title=batman)"
```

of course both are also mixable
```json
["something > 1 AND genre=comedy", "genre=horror OR title=batman"]
```

File: src/search.rs [hunk 2]
```diff
@@ -222,15 +217,8 @@ impl<'a> Query<'a> {
...
-        self.filters = Some(filters);
-        self
-    }
-    pub fn with_facet_filters<'b>(
-        &'b mut self,
-        facet_filters: &'a [&'a [&'a str]],
-    ) -> &'b mut Query<'a> {
-        self.facet_filters = Some(facet_filters);
+    pub fn with_filter<'b>(&'b mut self, filter: &'a str) -> &'b mut Query<'a> {
+        self.filter = Some(filter);
```
Comment (bidoubiwa) [inline src/search.rs:221]: Should be updated based on type update.

File: src/search.rs [hunk 3]
```diff
@@ -364,53 +352,27 @@ mod tests {
...
-
-        let mut query = Query::new(&index);
-        query.with_facet_filters(&[&["kind:text", "kind:title"]]);
-        let results: SearchResults<Document> = index.execute_query(&query).await.unwrap();
-        assert_eq!(results.hits.len(), 10);
-
-        client
-            .delete_index("test_query_facet_filters")
-            .await
-            .unwrap();
```
Comment (bidoubiwa) [inline src/search.rs:413]: Could you do these tests on `filter` ?

File: src/search.rs [hunk 4]
```diff
@@ -434,7 +396,7 @@ mod tests {
 
         let mut query = Query::new(&index);
         query.with_facets_distribution(Selectors::Some(&["kind"]));
-        query.with_facet_filters(&[&["kind:text"]]);
+        //query.with_facet_filters(&[&["kind:text"]]);
```
Comment (bidoubiwa) [inline src/search.rs:399]: should be uncommented

File: src/progress.rs [hunk 5]
```diff
@@ -200,18 +200,29 @@ pub struct SettingsUpdate {
...
+        #[serde(skip_serializing_if = "Option::is_none")]
+        number: Option<usize>,
+    },
+    DocumentsDeletion {
+        #[serde(skip_serializing_if = "Option::is_none")]
+        number: Option<usize>,
+    },
+    Settings {
+        settings: SettingsUpdate,
+    },
```
Comment (bidoubiwa) [inline src/progress.rs:225]: Thanks a lot for raising this change, was this documented somewhere?

I have made an issue to know if this is the expected behavior or not: see meilisearch/MeiliSearch/issues/1517

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
Hello, thanks so much for this pr 🔥 Small changes that were not included in this PR: 

- `settings.rs` lines 298, 495, 698 `[attributes for faceting]` in the comments should be changed with `filterable attributes`
- `settings.rs` lines 310, 513, 715 should have their API route updated `&format!("{}/indexes/{}/settings/attributes-for-faceting", self.host, self.uid),` to `/filterable-attributes`
- `.code-samples.meilisearch.yaml` 511, 535 `with_facet_filters` should be removed

File: src/search.rs (hunk 1 above)
Comment (bidoubiwa) [reply src/search.rs:150]: After some thought we may also just only accept the string format. What do you think?

**Comment by curquiza** [comment]
Since we don't have answer of the contributor, I merge and I will apply the changes myself :)

**Pull request merged**